src/GafferArnoldUI/ArnoldLightVisualiser.cpp

```diff
@@ -362,6 +436,30 @@ ArnoldLightVisualiser::~ArnoldLightVisualiser()
 {
 }
 
+Visualisations ArnoldLightVisualiser::visualise( const IECore::InternedString &attributeName, const IECoreScene::ShaderNetwork *shaderNetwork, const IECore::CompoundObject *attributes, IECoreGL::ConstStatePtr &state ) const
+{
+	Visualisations v = StandardLightVisualiser::visualise( attributeName, shaderNetwork, attributes, state );
+
+	if( shaderNetwork->outputShader()->getName() == "photometric_light" )
+	{
+		const CompoundData *shaderParameters = shaderNetwork->outputShader()->parametersData();
+		if( const StringData *iesFilenameData = shaderParameters->member<StringData>( "filename" ) )
+		{
+			IECoreGL::RenderablePtr iesVis = iesVisualisation( iesFilenameData->readable() );
+			if( iesVis )
+			{
+				IECoreGL::Renderable *geomVis = boost::const_pointer_cast<IECoreGL::Renderable>( v[VisualisationType::Geometry] ).get();
+				if( IECoreGL::Group *g = dynamic_cast<IECoreGL::Group *>( geomVis ) )
+				{
+					g->addChild( iesVis );
+				}
```

Comment:
So if we don't find the group we're expecting, we just silently don't display anything? Seems like we should either assume the group will always be there (that's a pretty reasonable assumption I think) or make our own group if none exists.

Reply:
Improved in https://github.com/GafferHQ/gaffer/pull/3531/commits/1bd6edbf77d7a826ada4560a719a6b28ea3b4950.